### DISCUSSION TITLE
Water limitation scaling, btran and where it gets applied

### DISCUSSION THREAD

**Discussion by walkeranthonyp** [NGEET/fates#1322] [General]
Apologies if this is already an open issue, I searched for our previous stomatal conductance (gs) conversations but I didn't find it. Following up on discussion about the `btran` wetness factor variable and how is get applied to leaf physiology. In non-hydro code btran is set as a function of soil water potential, and in hydro code (as I understand it) btran is a function of leaf water potential.  

btran is [applied to the intercept of the stomatal conductance function eg](https://github.com/NGEET/fates/blob/8827a6ea77f0e4f5d8408ddd7eeaa776bb3bba8a/biogeophys/FatesPlantRespPhotosynthMod.F90#L403 ):
```
 stomatal_intercept_btran = max( cf/rsmax0,stomatal_intercept(ft)*currentCohort%co_hydr%btran )
```

and [to vcmax (but not Jmax)](https://github.com/NGEET/fates/blob/8827a6ea77f0e4f5d8408ddd7eeaa776bb3bba8a/biogeophys/FatesPlantRespPhotosynthMod.F90#L1938-L1941):
```
     ! Adjust for water limitations 
      vcmax = vcmax * btran
      
      return
```

There are a number of issues that I think are worth discussing here. 1) application to stomatal intercept, 2) application to vcmax and not jmax, 3) btran dynamic with leaf water potential over the course of a day. I have to say I'm not super up to date with the literature in this space but I have kept some of an eye on it in the past.  

1) Application to stomatal intercept. AFAIK this is unique to the CLM family of models. It seems conceptually out on a limb as this term is supposed to represent a cuticular conductance that is beyond the control of plants, at least on physiological timescales. Applying btran to the intercept allows gs (or let's say leaf conductance) to go to zero when btran is zero, which seems unrealistic. 

    - 1a) I know @danicalombardozzi has done some work on this re night-time conductance and I would advocate for separating those two things within the model if we want to think more about night-time gs. Conceptually they are different, one being cuticular conductance the other nighttime conductance which is under plant control.    

2) Vcmax and Jmax are highly coupled, if Vcmax is scaled by btran, Jmax should also be.

3) I'm concerned that daily fluctuations of btran caused by fluctuations in leaf water potential double the impact of dry conditions on gs as gs is scaled for VPD/RH in the gs model, and then further scaled for leaf water potential. This could lead to the runaway shutdown of stomata and severely limit transpiration and carbon assimilation leading to dead plants. I'm not familiar with the literature here, but I have a hunch that pre-dawn leaf water potential might be a better factor to use to set btran. I'm not sure if pre-dawn vs time of measurement water potential has been tested in the literature.

And a couple more thoughts while I'm on a role.

4) Should btran be applied to some degree to respiration terms?

5) I advocate for removing the stomatal intercept term from the quadratic calculation of gs and use it simply as a minimum conductance when that calculation of gs < g0. If I understand right, for the Medlyn model at least, the stomatal slope is fit to the data assuming g0 = 0. Also, considering the gs model is based on an optimality concept, it could be assumed that the optimality would incorporate/account for leakage (I'm not explaining that well but I hope you see what I'm trying to get at). That would simplify the code some, with minimal impact to results.

**Comment by ckoven** [comment]
Hi @walkeranthonyp , these are all good points/questions.  Seems like it'd be good to test the effect of these out in some well-characterized testbeds to answer?  To point 3 in particular, I guess the question is whether midday depression in GPP & ET is driven more by the high midday VPD or the low midday psi_leaf? I was under the impression that it was tough to get the right amount of midday depression without taking into account the diurnal variation in psi_leaf, but maybe I'm mistaken on that? Do we have a clear set of observations that would let us test that?

**Comment by alistairrogers** [comment]
Hi Guys, we have a data set to test midday depression from Panama. Note that we've been working on issues surrounding these topics, As part of her paper Cherry is looking at both btran effects on vcmax and stomatal slope and comparing model output to diurnal gas exchange results from Panama. Anthony I apologise that I have not set up that stomatal conductance meeting yet. Perhaps this is morphing into a bigger discussion to include the points you raised above. The plan was to have a 2h mtg to discuss BNL work on gs including both Cherry's work and papers in prep from Julien and Ken that address issues with measurement and parameterization of stomatal slope and intercept, Anthony's thoughts on gs and Chonggang's issue with minimum conductance (?). Charlie, would you like to join too?

**Comment by ckoven** [comment]
thanks @alistairrogers, yes that sounds great.

**Comment by danicalombardozzi** [comment]
Hi Everyone:

Just to chime in here on several points that Anthony makes:

1) Related to application/use of btran:
You are correct in that (at least before PHS) CLM applied btran to the stomatal intercept and to Vcmax. I also agree that there are major problems associated with this representation of water stress. Applying btran to any other part of the stomatal conductance code would effectively double-count the impact of water stress on stomatal conductance since it is applied to photosynthesis and then photosynthesis is used to calculate stomatal conductance. 

Given that Dan Kennedy @djk2120 implemented the PHS model, I assumed that the water stress calculation in CLM has changed, although I haven't looked very closely. I think it's worth revisiting this to a) confirm whether btran is still being applied to Vcmax and g0 with PHS (perhaps its set to 1 in this case?); and b) if so, test removing btran from g0 calculation and adding it to Jmax.

2) Related to stomatal intercept:
The reason that the intercept term is included in the Medlyn model is because the model will crash if conductance goes to 0. The intercept was therefore set to a very small number to avoid crashing the model. 

As far as cuticular and nighttime conductance are concerned: I would not advocate for using the intercept term as a nighttime conductance value, and think instead that the intercept is more related to cuticular conductance. There is also a difference between nighttime conductance and minimum conductance. In my GMD paper, we used observations of nighttime conductance to constrain conductance only during night (when incoming solar was 0) and also to constrain minimum stomatal conductance, finding that the minimum daytime conductance often dropped below observations of nighttime conductance. At the time, there wasn't enough empirical evidence to suggest whether or not this was in fact was realistic. 

I'd advocate for potentially keeping -- although probing the sensitivity of -- the stomatal slope term, as it will save from errors related to gs=0 and can theoretically be considered cuticular conductance. However, I agree that perhaps it could be removed from the quadratic solver. It's definitely worth investigating further since the Medlyn model does not include a slope term, and there is not strong observational evidence to quantify whether this may or may not be representative of cuticular conductance. From a physiological perspective, however, it seems likely that plants may lose some water even with stomata closed, and gs=0 (e.g., stomata close perfectly) may not be realistic under all scenarios.

**Comment by djk2120** [comment]
CLM45 has btran applied to g0, but that was removed for CLM5. If I recall correctly, that was changed when the Medlyn model was introduced. CLM5 btran is indeed applied to vcmax (and not to jmax or g0). Happy to discuss further and/or try out other ideas. 

In terms of the dual limitations imposed by Medlyn and hydraulic stress, to me this is very much an open question in the literature. I do like downregulation based on leaf water potential, because it seems like plants must avoid embolism which is better described by midday LWP than by predawn LWP (ie soil potential). The idea would be that Medlyn downregulates gs with some optimality idea in mind. Then if the LWP required to provide that sap flux is too negative, then it might be downregulated further by Btran, reflecting the need to avoid embolism. Importantly you'll note that the hydraulics should 'see' the Medlyn downregulation, and only downregulate further if needed. I think that understanding how plants go about maximizing carbon gain subject to hydraulic constraints is important and will hopefully become clearer over time. But that's essentially what we were going for by layering together Medlyn and Btran. We are certainly open to fine-tuning this approach in the future, as the literature homes in on a solution.

To (4), btran is applied to leaf maintenance respiration in the CLM big leaf models.

**Comment by tompowell9** [comment]
WRT to point 3...
I was curious about this issue as well when I first started using ED2-hydro several years ago.  I have not explicitly tested for this dynamic with FATES-hydro, but I suspect it will behave similar to ED2-hydro because ED2-hydro also double dips on btran--i.e. gs is first down-regulated by VPD, and then the 0-1 beta factor is applied to gs as a function of LWP.  

Dan's description--"only down regulate further if needed"--is essentially how this wonky representation of gs in ED2-hydro behaves.  I found that when beta is moved from the soil to the leaf, it causes the hysteresis of gs to be stronger. But as it turns out, it is difficult for this formulation of gs to drive run-away stomatal closure because as LWP down-regulates gs, it also results in less water being deducted from the supply. Hence, the supply remains intact to recharge the capacitor overnight and makes water available for use the next morning so that the stomata can be fully open. The model doesn't capture these dynamics when beta is a function of psi_soil. Rather, when beta is a function of psi_soil, beta is more even throughout the day and the soil moisture supply can become over-drawn such that beta (and hence gs) starts the day in a highly suppressed state (not shown in figure below, See Powell et al. 2013 NP, Figure 10). 
  
![image](https://user-images.githubusercontent.com/10078952/107449674-d735e800-6af8-11eb-999c-c11fe0f235a8.png)

**Comment by walkeranthonyp** [comment]
All good points. Just to be clear @danicalombardozzi I'm not advocating to remove g0 entirely which could allow GS to go to zero as you say. Just that it could be used simply as a minimum and not as an additional term in the GS function. That would simplify the code and would be more in line with how G1 is estimated from data.

**Comment by bishtgautam** [comment]
- Medlyn model + hydraulic stress: One could combine the Medlyn model with the [Manzoni model](https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/j.1365-2435.2010.01822.x) i.e. the optimization factor `lambda` is a function of leaf water potential. 

- Though, [Wang et al. (2020)](https://nph.onlinelibrary.wiley.com/doi/full/10.1111/nph.16572) argue that both Medlyn and Manzoni models do not satisfy all the criteria tests that they present in their study.

**Comment by walkeranthonyp** [comment]
Thanks for the links @bishtgautam. There's also this by [Joshi et al (preprint)](https://www.biorxiv.org/content/10.1101/2020.12.17.423132v1) in Prentice's lab. I've not read it but I'm guessing they're doing something along the lines of what you suggest in your first bullet

**Comment by danicalombardozzi** [comment]
> All good points. Just to be clear @danicalombardozzi I'm not advocating to remove g0 entirely which could allow GS to go to zero as you say. Just that it could be used simply as a minimum and not as an additional term in the GS function. That would simplify the code and would be more in line with how G1 is estimated from data.

@walkeranthonyp Yes, using g0 as a minimum, rather than adding it to the equation at all times, seems reasonable to me.

**Comment by serbinsh** [comment]
@alistairrogers mentioned this thread in the meeting today but I didn't see the list so posting this here

Link to notes: https://docs.google.com/document/d/192to2luT7ETsFxNwFOxAABS_7951ILWoXBwpjCL06Nk/edit?usp=sharing
